core/templates/pages/topics-and-skills-dashboard-page/topics-list/topics-list.component.html

```diff
@@ -152,6 +276,91 @@
+              <span *ngIf="!topic.classroom">Unassigned</span>
+            </div>
+          </div>
+          <div class="topic-item-options">
+            <button class="fas fa-ellipsis-v topic-edit-box-icon" role="button" aria-label="Edit Options" (click)="changeEditOptions(topic.id)"></button>
+            <div *ngIf="showEditOptions(topic.id)" class="topic-edit-options">
+              <a class="topic-edit-options-edit" [href]="getTopicEditorUrl(topic.id)">
+                <i class="fa fa-pen"></i>
+                <span>Edit</span>
+              </a>
+              <a *ngIf="userCanDeleteTopic" (click)="deleteTopic(topic.id, topic.name)">
+                <i class="fas fa-trash-alt"></i>
+                <span>Delete</span>
+              </a>
+            </div>
+          </div>
+        </div>
+        <div class="topic-item-topic-status">
+          <span class="topic-item-topic-status-header">Status</span>
+          <span *ngIf="topic.isPublished && topic.getTotalChaptersCounts().length && topic.getTotalChaptersCounts().length == fullyPublishedStoriesCounts[idx]">
+            Fully Published
+          </span>
+          <span *ngIf="topic.isPublished && (topic.getTotalChaptersCounts().length==0 || topic.getTotalChaptersCounts().length !== fullyPublishedStoriesCounts[idx])">
+            Partially Published
+          </span>
+          <span *ngIf="!topic.isPublished">
+            Not Published
+          </span>
+        </div>
+        <div class="topic-item-topic-description">
+          <span class="topic-item-description-header">Description: </span>
+          <span>{{ topic.description }}</span>
+        </div>
+        <div class="topic-item-entities">
+          <div class="topic-item-topic-entity published-stories-entity">
+            <div class="topic-item-topic-entity-header">
+              Published Stories
+            </div>
+            <span *ngIf="partiallyPublishedStoriesCounts[idx]">
+              {{ partiallyPublishedStoriesCounts[idx] }} Partial
+              <br>
+              (<span>{{ publishedChaptersInPartiallyPublishedStories[idx] }} published</span>
+              <br>
+              <span>{{ totalChaptersInPartiallyPublishedStories[idx] }} total</span>)
+            </span>
+            <span *ngIf="fullyPublishedStoriesCounts[idx]">
+              {{ fullyPublishedStoriesCounts[idx] }} Full
+            </span>
+            <span *ngIf="!partiallyPublishedStoriesCounts[idx] && !fullyPublishedStoriesCounts[idx]"
+                  class="text-center">
+              --
+            </span>
+          </div>
+          <div class="topic-item-topic-entity">
+            <div class="topic-item-topic-entity-header">
+              Chapter Publication Notifications
+            </div>
+            <span *ngIf="topic.getTotalUpcomingChaptersCount()==1"> {{ topic.getTotalUpcomingChaptersCount() }} upcoming launch in the next 14 days</span>
```

Comment:
Ditto as above.

Also I am a bit concerned about the repetition of the logic here. I suggest you create a TS method that gives you the text for this span, and use it in both places. Please try not to copy/paste logic into multiple parts of the code, since it makes maintenance difficult.

Reply:
Done.